extensions/libarchive/BinFiles.cpp

```diff
@@ -97,8 +97,8 @@ void BinFiles::preprocessFlowFile(core::ProcessContext* /*context*/, core::Proce
 void BinManager::gatherReadyBins() {
   std::lock_guard < std::mutex > lock(mutex_);
   std::vector< std::string > emptyQueue;
-  for (std::map<std::string, std::unique_ptr<std::deque<std::unique_ptr<Bin>>> >::iterator it=groupBinMap_.begin(); it !=groupBinMap_.end(); ++it) {
-    std::unique_ptr < std::deque<std::unique_ptr<Bin>>>&queue = it->second;
+  for (auto& kvp : groupBinMap_) {
+    auto &queue = kvp.second;
```

Comment:
This is the perfect use case for structured bindings.
```suggestion
  for (auto& [key, queue] : groupBinMap_) {
```

Reply:
Good catch, updated in c067c00af8bb65f077173a4398705e59aec59339